mobile/native/src/trade/position/handler.rs

```diff
@@ -43,34 +41,50 @@ pub fn get_position_matching_order(order: &Order) -> Result<Option<Position>> {
     }
 }
 
-/// Sets the position to the given state
+/// Set the position to the given [`PositionState`].
 pub fn set_position_state(state: PositionState) -> Result<()> {
     if let Some(position) = db::get_positions()?.first() {
-        db::update_position_state(position.contract_symbol, state)?;
-        let mut position = position.clone();
-        position.position_state = state;
+        let position = db::update_position_state(position.contract_symbol, state)?;
         event::publish(&EventInternal::PositionUpdateNotification(position));
     }
 
     Ok(())
 }
 
-/// A channel renewal could be triggered for:
+/// A channel renewal could be triggered to:
 ///
-/// - Rolling over (no offer associated).
-/// - Opening a new position.
-/// - Resizing a new position.
+/// - Roll over (no offer associated).
+/// - Open a new position.
+/// - Resize a position.
 pub fn handle_channel_renewal_offer(expiry_timestamp: OffsetDateTime) -> Result<()> {
-    // FIXME: This won't always be true once we reintroduce position resizing.
     if let Some(position) = db::get_positions()?.first() {
-        tracing::debug!("Setting position to rollover");
-        db::rollover_position(position.contract_symbol, expiry_timestamp)?;
-        let mut position = position.clone();
-        position.position_state = PositionState::Rollover;
-        position.expiry = expiry_timestamp;
-        event::publish(&EventInternal::PositionUpdateNotification(position));
+        // Assume that if there is an order in filling we are dealing with position resizing.
+        //
+        // TODO: This is garbage as we very well know. Any other ideas? We could generate
```

Comment:
@bonomat and I were talking before, that it would probably make sense to add a custom message to wrap the renew protocol for rollover, resizing and async matching.. and enrich the message with some 10101 meta data. that would also allow us to get rid of the ugly websocket communication to send over some additional data.

Reply:
That sounds good, but I think that belongs in a different PR. Given your approval, I assume you are okay with not tackling this TODO right now.